react-components/jsx.md

```diff
@@ -0,0 +1,321 @@
+          Edit <code>src/App.js</code> and save to reload.
+        </p>
+        <a
+          className="App-link"
+          href="https://reactjs.org"
+          target="_blank"
+          rel="noopener noreferrer"
+        >
+          Learn React
+        </a>
+      </header>
+    </div>
+  );
+}
+```
+
+The `App` function returns one JSX object. This JSX object opens with a `<div>` element.
+
+This `<div>` element has one child: the `<header>` element. The `<header>` element has three children: `<img>`, `<p>`, and `<a>`.
+
+### !callout-info
+
+## Modifying `App` to Use Variables
+
+We can modify our `App` function to use more variables. The JSX object can be defined inside the function, and then returned.
+
+```js
+function App() {
+  const appElements = <div className="App">... more elements ...</div>;
+  return appElements;
+}
+```
+
+Here, we are creating the `const` variable `appElements`, setting it to a JSX object, and then returning it.
+
+### !end-callout
+
+## Embedding Expressions in JSX
+
+When we have a JavaScript expression, such as a variable or function call, to add into a JSX object, we can do so with `{}`.
+
+```js
+const guestName = "Megha";
+
+const welcomeMessage = (
+  <section>
+    <h1>Welcome {guestName}~</h1>
+    <p>It's nice to hear from you again!</p>
+  </section>
+);
+```
+
+In this situation, the rendered HTML would look like:
+
+```html
+<section>
+  <h1>Welcome Megha~</h1>
+  <p>It's nice to hear from you again!</p>
+</section>
+```
+
+### Embedding Expressions: `App.js`
+
+Let's revisit this `src/App.js` again!
+
+```js
+import logo from "./logo.svg";
+import "./App.css";
+
+function App() {
+  return (
+    <div className="App">
+      <header className="App-header">
+        <img src={logo} className="App-logo" alt="logo" />
+        <p>
+          Edit <code>src/App.js</code> and save to reload.
+        </p>
+        <a
+          className="App-link"
+          href="https://reactjs.org"
+          target="_blank"
+          rel="noopener noreferrer"
+        >
+          Learn React
+        </a>
+      </header>
+    </div>
+  );
+}
+```
+
+We can focus on this part of the JSX:
+
+```js
+<img src={logo} className="App-logo" alt="logo" />
+```
+
+The value of the `src` attribute for this element will be the value of `logo`. `logo` was imported from `./logo.svg` at the top of `App.js`!
+
+## Element Attributes in JSX Objects
+
+We can define any attributes for any element in a JSX object. We use syntax that is nearly identical to what we'd use for HTML elements.
+
+```js
+<a href="https://reactjs.org" target="_blank" rel="noopener noreferrer">
+  Learn React
+</a>
+```
+
+This `<a>` element will have the following HTML attributes when it's rendered:
+
+- `href`
+- `target`
+- `rel`
+
+## CSS Classes in JSX Objects
+
+One of the most important differences between JSX objects and HTML elements is how to add classes.
+
+In JSX, we use the attribute `className` to define `class` in rendered HTML.
```

Comment:
make sure we're clear about what kind of classes we're discussing
the reader is familiar with HTML class attributes, so lead with that
```suggestion
One of the most important differences between JSX objects and HTML elements is how to add CSS classes.

While HTML elements use the `class` attribute, in JSX we use the `className` attribute.
```